iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant_internal.hpp

```diff
@@ -98,6 +98,8 @@ struct call_at_index
     {
         if (N == index)
         {
+            // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type safety ensured through template parameter
+            // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
             reinterpret_cast<T*>(ptr)->~T();
```

Comment:
@elfenpiff Similar to the `cxx::vector` the usage of `reinterpret_cast` makes it impossible to use the `variant` as `constexpr` as `reinterpret_cast` can only be executed during runtime. This might be something for iceoryx 4.0 :smile:

Reply:
You might mix up `dynamic_cast` and `reinterpret_cast`. The former is runtime but the latter is compile time. The big Problem for `constexpr` is the destructor. But you are right, maybe something for v4.0

Reply:
@elBoberido Nope, `reinterpret_cast` is not possible during compile-time, see [here](https://stackoverflow.com/a/59913380/18276160).

Reply:
@mossmaurice hahaha, you are right but I have the support of a buggy compiler :sweat_smile: